### DISCUSSION TITLE
Using react components

### DISCUSSION THREAD

**Discussion by c6p** [observablehq/framework#1095] [Q&A] [answered]
```js
import { createElement } from 'npm:react';
import { createRoot } from 'npm:react-dom/client';
import { MultiSelect } from 'npm:@mantine/core';
display(createRoot)
```
Displays `TypeError: error loading dynamically imported module: http://127.0.0.1:3000/_npm/react-dom@18.2.0/client`.

**Comment by c6p** [comment, marked as answer]
**Updated.**

Define `root` div to mount our component. Note: `selected` is reactive. Yay!
```md
<link rel="stylesheet" href="npm:@mantine/core/styles.css" />
<div id="root"></div>

Selected values: ${selected.join(', ')}
```
I've found out easiest way to manage state via `Mutable`. Here we define a reusable `useState` hook as in react.
```js
import {Mutable} from "npm:@observablehq/stdlib";
function useState(value) {
  const state = Mutable(value)
  const setState = (value) => state.value = value;
  return [state, setState]

}
const [selected, setSelected] = useState(["react"])
```

I misunderstood how observable renders the page. It was enough to just import them as you would in node.
Since JSX is not supported, use `createElement`.
```js
import {createElement} from 'npm:react';
import {createRoot} from 'npm:react-dom';
import { MultiSelect, MantineProvider } from 'npm:@mantine/core';
const root = createRoot(document.getElementById('root'));
root.render(
    createElement(MantineProvider, {},
        createElement(MultiSelect, {
          label: "your favorite libraries",
          placeholder: "pick value",
          data: ['react', 'angular', 'vue', 'svelte'],
          value: selected,
          clearable: true,
          searchable: true,
          onChange: setSelected
        })
    )
)
```

**Reply by Fil** [reply]
Nice! Can you share a more complete example?

**Reply by c6p** [reply]
Updated the answer with a complete example.

**Reply by Fil** [reply]
`<link rel="stylesheet" href="npm:@mantine/core/styles.css" />` seems to work—it copies the css file and self-hosts it under `http://127.0.0.1:3000/_npm/@mantine/core@7.6.2/styles.css`) (using Framework@1.4.0 which was released yesterday).

**Reply by c6p** [reply]
You are right! After a npm update, it works.
